### ISSUE TITLE
[IMPROVE] Fields with empty title shall have exactly the same value as the user entered in the name field, without altering it

### ISSUE THREAD

**Issue by damianpumar** [recognai/rubrix#3625]
When a user creates a new dataset, he has the possibility to create fields with empty Title.

Currently, this title is being set to the same value as the name but Capitalized format from SDK code. This logic should be moved to the **backend** because the **frontend** should do exactly the same thing that the **SDK** do.

This is necessary to unify the business logic in one place.

Tasks:
- Move the SDK logic as-is to the backend.
- Remove the validation that the backend has when processing the field request, as the title may be optional.
- Think about what should be the correct format of the title, as ideally the value entered by the user should NOT be modified.

Examples: 
- If the user enters a field with name: FOO, the title should be FOO.
- If the user enters a field with name: foo, the title should be foo.
- If the user enters a field with name: fOo, the title must be fOo.
In short, the value the user entered should not be changed.
